src/main/java/TaskHelper.java

```diff
@@ -0,0 +1,121 @@
+import java.util.ArrayList;
+
+public class TaskHelper {
+    private static TaskHelper INSTANCE = null;
+    private final ArrayList<Task> taskList = new ArrayList<>();
+
+    public static TaskHelper getInstance(){
+        if(INSTANCE == null){
+            INSTANCE = new TaskHelper();
+        }
+        return INSTANCE;
+    }
+
+    private static void printHorizontalLine(){
+        System.out.println("-----------------------------------------------");
+    }
+
+    private static void printEmptyLine(){
+        System.out.println();
+    }
+
+    public void printAllTasks(){
+        if(taskList.size() == 0) {
+            System.out.println("You have not added any tasks yet!");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.size(); i++) {
+                int itemIdx = i + 1;
+                Task task = taskList.get(i);
+                System.out.println(itemIdx + ". " + task);
+            }
+        }
+    }
+
+    public void addTodo(String commandArgs){
+        Todo todo = new Todo(commandArgs);
+        taskList.add(todo);
+        printAddTodoMsg(todo);
+    }
+
+    public void addDeadline(String commandArgs){
+        final String matchByPrefix = "/by";
+        final int indexOfByPrefix = commandArgs.indexOf(matchByPrefix);
+        String description = commandArgs.substring(0, indexOfByPrefix).trim();
+        String by = commandArgs.substring(indexOfByPrefix).replace("/by", "").trim();
+        Deadline deadline = new Deadline(description, by);
+        taskList.add(deadline);
+        printAddDeadlineMsg(deadline);
+    }
+
+    public void addEvent(String commandArgs){
+        final String matchByPrefix = "/at";
+        final int indexOfByPrefix = commandArgs.indexOf(matchByPrefix);
+        String description = commandArgs.substring(0, indexOfByPrefix).trim();
+        String at = commandArgs.substring(indexOfByPrefix).replace("/at", "").trim();
+        Event event = new Event(description, at);
+        taskList.add(event);
+        printAddEventMsg(event);
+    }
+
+    private void printAddTodoMsg(Todo todo){
+        printHorizontalLine();
+        System.out.println("added: " + todo);
+        if(taskList.size() == 1){
+            System.out.println("You have " + taskList.size() + " task in the list");
+        } else {
+            System.out.println("You have " + taskList.size() + " tasks in the list");
+        }
+        printEmptyLine();
+        printHorizontalLine();
+    }
```

Comment:
Not a coding standard violation but... it seems to me that your printAddTodoMsg, printAddDeadlineMsg and printAddEventMsg all do the same thing, I think you can consider putting it all under one method and just changing the method parameters to accept a Task type member variable, then you can use it for all different cases and also shorten your code! :)

Reply:
I really loved this suggestion. Thank you! Did not realize that I could do that before. Appreciate that!